.github/workflows/build-all.yml

```diff
@@ -16,33 +21,113 @@ jobs:
+          popd
+
+      - name: Copy current workflow artifacts to artifact folders
+        run: |
+          pushd downloaded-artifacts
+          pwd
+          ls -la
+          for f in $(ls); do
+            mv $f/*.sha256sum ../artifact-shas/.
+            mv $f/* ../artifact-binaries/.
+          done
+          popd
+
+      - name: Check SHAs
+        run: |
+          cd artifact-binaries
+          ls ../artifact-shas/*.sha256sum | xargs sha256sum --check
+
+      - name: Generate SHA summary
+        id: generate_sha_file
+        run: |
+          echo "---" >> $GITHUB_STEP_SUMMARY
+          echo "Update $(date -u +%F) $(date -u +%T) GMT/UTC" >> $GITHUB_STEP_SUMMARY
+          echo "### SHAs of produced and carried forward binaries by this workflow" >> $GITHUB_STEP_SUMMARY
+          echo "  - $GITHUB_SERVER_URL/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID" >> $GITHUB_STEP_SUMMARY
+          echo '```' >> $GITHUB_STEP_SUMMARY
+          sha_output_file=${GITHUB_SHA}_${RANDOM}_shas.txt
+          echo "sha_output_file=${sha_output_file}" >> $GITHUB_OUTPUT
+          cat artifact-shas/*.sha256sum > ${sha_output_file}
+          cat artifact-shas/*.sha256sum >> $GITHUB_STEP_SUMMARY
+          echo '```' >> $GITHUB_STEP_SUMMARY
+          # Get a random string of characters to represent the EOF delimiter
+          EOF=$(dd if=/dev/urandom bs=15 count=1 status=none | base64)
+          echo "sha_summary<<$EOF" >> $GITHUB_ENV
+          cat $GITHUB_STEP_SUMMARY >> $GITHUB_ENV
+          echo "$EOF" >> $GITHUB_ENV
+
+      - name: Determine release tag to upload assets to and draft type
+        run: |
+          if [[ "${{ inputs.newRelease }}" == "false" ]]; then
+            echo "use_release_tag=${{ steps.get_previous_release.outputs.prev_release }}" >> $GITHUB_ENV
+            echo "create_draft=false" >> $GITHUB_ENV
+          else
+            echo "use_release_tag=draft_release_${{ github.sha }}" >> $GITHUB_ENV
+            echo "create_draft=true" >> $GITHUB_ENV
+          fi
+
+      - name: Add binaries to release
+        id: create_release
+        uses: softprops/action-gh-release@d4e8205d7e959a9107da6396278b2f1f07af0f9b
         with:
-          workflow: build-windows.yml
-          path: artifacts
-      - run: ls -laR artifacts
+          token: ${{ github.token }}
+          draft: ${{ env.create_draft }}
+          tag_name: ${{ env.use_release_tag }}
+          files: |
+            artifact-shas/*
+            artifact-binaries/*
+          body: "${{ env.sha_summary }}"
+          generate_release_notes: true
+          append_body: true
+
+      - name: Add release url to summary
+        run: echo "Release created/updated at ${{ steps.create_release.outputs.url }}" >> $GITHUB_STEP_SUMMARY
+
+      - name: Print diff for future PR submission
+        run: |
+          yarn install
+          yarn build
+          yarn run updateExpected --input ${{ steps.generate_sha_file.outputs.sha_output_file }}
```

Comment:
I would do this in a separeted job, also you can use https://github.com/peter-evans/create-pull-request to create the new PR

Reply:
dropped the step for now